tests/test_send_file.py

```diff
@@ -159,7 +159,7 @@ def test_from_directory(directory, path):
     rv.close()
 
 
-@pytest.mark.parametrize("path", ["../unsafe.txt", "nothing.txt", "null\x00.txt"])
+@pytest.mark.parametrize("path", ["../res/test.txt", "nothing.txt", "null\x00.txt"])
```

Comment:
Why did this value change?

Reply:
With `../unsafe.txt` we don't know if it fails because of unsafe path or because the file isn't exist (you can do a simple experiment, comment [those lines](https://github.com/pallets/werkzeug/blob/cf9ebfc2611679b6fb089f11edeeb4d50672077a/src/werkzeug/security.py#L217-L223) in `safe_join` and run test). New value points to the real file, so if unsafe path isn't filtered out, it returns a file and test fails.

Reply:
@davidism What do you think?

Reply:
I think that makes sense. I would also keep the `../unsafe.txt` test as well (as I understand things that is also worth testing given the `../` prefix). Could you split this into two commits, the first fixing the path usage and the second adding the new test?

Reply:
I can do it, but what would you test with `../unsafe.txt`? As long as this file doesn't exist, a test with path to it will pass anyway - it's the same as `nothing.txt`. If you think the test should be more readable, we can just add `unsafe.txt` to the repo instead of my change to `../res/test.txt`?

Reply:
I think the `../unsafe.txt` can act as a regression test to ensure that [this aspect](https://github.com/pallets/werkzeug/blob/dc4169b13a131a6cd68a756a4430b1945727d05e/src/werkzeug/security.py#L241) works as expected. Unless I'm confused I think the `../` prefix check is before the check if the path exists.

The above said, I now see this test [exists](https://github.com/pallets/werkzeug/blob/dc4169b13a131a6cd68a756a4430b1945727d05e/tests/test_security.py#L69) which negates the point of testing `../unsafe.txt` in this test. I'll merge this.